magicctapipe/io/io.py

```diff
@@ -92,9 +134,49 @@ def format_object(input_object):
     string = string.replace("'", "").replace(",", "")
 
     return string
+    
+def get_stereo(
+    event_data, config, quality_cuts=None, group_index=["obs_id", "event_id"]
+):
+
+    """
+    Gets the stereo events surviving specified quality cuts without overwriting the
+    multiplicity and combo type. This function is useful when loading the DL2 data
+    when we apply one RF per telescope (and not per combo type).
+
+    Parameters
+    ----------
+    event_data: pandas.core.frame.DataFrame
+        Data frame of shower events
+    config: dict 
+        Read from the yaml file with information about the telescope IDs. Typically called "config_general.yaml"
+    quality_cuts: str
+        Quality cuts applied to the input data
+    group_index: list
+        Index to group telescope events
+    
 
+    Returns
+    -------
+    event_data_stereo: pandas.core.frame.DataFrame
+        Data frame of the stereo events surviving the quality cuts
+    """
+    
+    TEL_NAMES, TEL_COMBINATIONS = telescope_combinations(config)
 
-def get_stereo_events(
+    event_data_stereo = event_data.copy()
+
+    # Apply the quality cuts                                                                                                                                          
+    if quality_cuts is not None:
+        event_data_stereo.query(quality_cuts, inplace=True)
+
+    max_multiplicity=len(TEL_NAMES.keys())
+    # Extract stereo events                                                                                                                                           
+    event_data_stereo["multiplicity"] = event_data_stereo.groupby(group_index).size()
+    event_data_stereo.query(f"multiplicity >1 & multiplicity <= {max_multiplicity}", inplace=True)
+    
+    return event_data_stereo
+def get_stereo_events_old(
```

Comment:
This function is needed by the steps RFs->DL3, which are the ones from the 'old' MCP, because in the new version we are reading telescope names and ids from a configuration file, so that we need to pass this configuration file to the io functions too. But in the 'old' version names and combos are fixed and the functions (and scripts!) do not ask for the config file. If you use get_stereo_events, e.g., in the Dl1 to DL2 script (not updated in this PR) the code will complain about the function arguments (the new get_stereo function needs 2 arguments, while the old one, used in this script, uses 1 argument)